ui/packages/catalog-ui-search/src/main/webapp/js/model/MetacardAction.js

```diff
@@ -11,17 +11,32 @@
  **/
 var Backbone = require('backbone');
 require('backbone-associations');
+const URL = require('url');
+const QS = require('querystring');
 
 module.exports = Backbone.AssociatedModel.extend({
     defaults: function() {
         return {
             url: undefined,
             title: undefined,
             description: undefined,
-            id: undefined
+            id: undefined,
+            queryId: undefined
         };
     },
     getExportType: function() {
         return this.get('title').replace(/^Export( as)?\s+\b/, '');
+    },
+    initialize: function() {
+        this.handleQueryId();
+        this.listenTo(this, 'change:queryId', this.handleQueryId);
+    },
+    handleQueryId: function() {
```

Comment:
❓ I wonder if there is a way to only add the query ID if the action requires it.  Thoughts?

Reply:
Updated to only add the query id if the action requires it.